src/vs/vscode.d.ts

```diff
@@ -1962,6 +1962,63 @@ declare namespace vscode {
 	}
 
 	/**
+	 * Reasons for which completion can be triggered.
+	 */
+	export enum CompletionTriggerReason {
+
+		/**
+		 * Invoked by the user, e.g. by hitting `Ctrl+Space`
+		 */
+		OnManualInvocation = 1,
+
+		/**
+		 * When a trigger character was types or when the configuration
+		 * is set to trigger on any character.
+		 */
+		OnType = 2
+	}
+
+	/**
+	 * Value-object that contains additional information when
+	 * requesting completions.
+	 */
+	export interface CompletionContext {
+		triggerReason: CompletionTriggerReason;
+		triggerCharacter: string;
+	}
+
+	/**
+	 * Represents a list of completions.
+	 */
+	export class CompletionList extends Array<CompletionItem> {
+
+		/**
+		 * The context from which this completion list was computed.
+		 */
+		context: CompletionContext;
```

Comment:
Yeah, questionable. It's sort of an open door for commands that receive this. Tho we should maybe not have it